qemu/qmp.go

```diff
@@ -1154,6 +1154,14 @@ func (q *QMP) ExecuteCPUDeviceAdd(ctx context.Context, driver, cpuID, socketID,
 		"thread-id": threadID,
 	}
 
+	if socketID != "" {
```

Comment:
Don't we need to remove the initialisation of socket-id and thread-id in the composite literal that initialises the map?  Otherwise, this patch doesn't actually do anything, apart from conditionally re-initialise keys that have already been initialised.

Reply:
@markdryan: Thanks for the review. Updated the patch.